core/src/main/java/org/elasticsearch/rest/action/exists/RestExistsAction.java

```diff
@@ -51,9 +58,11 @@ public void handleRequest(final RestRequest request, final RestChannel channel,
         if (RestActions.hasBodyContent(request)) {
             existsRequest.source(RestActions.getRestContent(request));
         } else {
-            QuerySourceBuilder querySourceBuilder = RestActions.parseQuerySource(request);
-            if (querySourceBuilder != null) {
-                existsRequest.source(querySourceBuilder);
+            QueryBuilder<?> queryBuilder = RestActions.parseQuerySource(request);
+            if (queryBuilder != null) {
+                QuerySourceBuilder querySourceBuilder = new QuerySourceBuilder();
+                querySourceBuilder.setQuery(queryBuilder);
+                existsRequest.source(querySourceBuilder.buildAsBytes());
```

Comment:
I have thought about that and want to do it, but wanted to try and keep this PR as small as possible (its already quite big) and refactor those request types in a different PR. wdyt?